File: cpp/bc2059.cpp
#include <iostream>

using namespace std;

int main()
{
	int ePla, fPla, sPla, ro, ac, so, winner = 0;
	
	cin >> ePla >> fPla >> sPla >> ro >> ac;
	
	so  = (fPla + sPla) % 2;
	
	if(ro == 0 && ac == 0)
	{
		if(ePla == 1)
		{
			if(ePla == 1)
				winner = so == 0? 1 : 2;
		}
		else
		{
			winner = so != 0? 1 : 2; 
		}
	}
	else 
	{
		if(ro == 1 && ac == 1) winner = 2;
		else if((ro == 1 && ac == 0) || (ro == 0 && ac == 1)) winner = 1;
	}
	
	cout << "Jogador " << winner << " ganha!" << endl;
}

File: python/bc2059.py
cVal = input().split(" ")

p = int(cVal[0])
j1 = int(cVal[1])
j2 = int(cVal[2])
r = int(cVal[3])
a = int(cVal[4])

winner = 0

if a == 1:
    winner = 2 if r == 1 else 1
elif r == 1:
    winner = 1 if a == 0 else 2
else:
    if p == 1:
        winner = 1 if (j1+j2) % 2 == 0 else 2
    else:
        winner = 1 if (j1+j2) % 2 != 0 else 2

print(f"Jogador {winner} ganha!")